src/common/snippets/src/lowered/pass/split_loops.cpp

```diff
@@ -0,0 +1,98 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "snippets/lowered/pass/split_loops.hpp"
+
+#include "snippets/lowered/linear_ir.hpp"
+#include "snippets/lowered/loop_manager.hpp"
+#include "snippets/snippets_isa.hpp"
+#include "snippets/itt.hpp"
+
+namespace ov {
+namespace snippets {
+namespace lowered {
+namespace pass {
+using LoopManager = LinearIR::LoopManager;
+using LoopInfoPtr = LoopManager::LoopInfoPtr;
+
+SplitLoops::SplitLoops() : Pass() {}
+
+bool SplitLoops::can_be_split(const LoopInfoPtr& current, const LoopInfoPtr& parent) {
+    return current->work_amount == parent->work_amount && current->dim_idx == parent->dim_idx &&
+           current->increment != parent->increment;
+}
+
+bool SplitLoops::run(LinearIR& linear_ir) {
+    OV_ITT_SCOPED_TASK(ov::pass::itt::domains::SnippetsTransform, "Snippets::SplitLoops")
+    if (linear_ir.empty())
+        return false;
+
+    const auto& loop_manager = linear_ir.get_loop_manager();
+    bool loop_was_split = false;
+    for (const auto& expr : linear_ir) {
+        const auto& loop_ids = expr->get_loop_ids();
+        if (loop_ids.empty())
+            continue;
+
+        // Note: we currently consider only the outermost loops for splitting
+        // Splitting could also be done in a more general case, but the splitted loop and its parent must always
+        // be in the same set of outer loops. Otherwise they won't be fused.
+        const auto& loop_id = loop_ids.front();
+        const auto loop = loop_manager->get_loop_info(loop_id);
+        for (const auto& entry_point : loop->entry_points) {
+            const auto& parent_port = entry_point.expr_port->get_port_connector_ptr()->get_source();
+            const auto& parent_expr = parent_port.get_expr();
+            const auto parent_loop_ids = parent_expr->get_loop_ids();
+            if (parent_loop_ids.empty())
+                continue;
+
+            const auto& parent_loop_id = parent_loop_ids.front();
+            const auto parent_loop_port = loop_manager->get_loop_port_by_expr_port(parent_port, parent_loop_id);
+            // We don't split loop which are not compatible with parent loop because such loops will not be fused
+            if (parent_loop_port.is_incremented != entry_point.is_incremented)
```

Comment:
As I understand, we can split Loop only when the fusing is possible, correct?
Could we create the public static method in `FuseLoops` and call it here? Could you please check if it's possible to create? I'd like to have this logic (when we can fuse and cannot) only in one place: `FuseLoops` pass.

Reply:
It was agreed offline that it's necessary to check only ports compatibility. `FuseLoops::loop_ports_are_compatible` has been made static and reused in this pass